bin/ss-docs-lint

```diff
@@ -0,0 +1,197 @@
+getopt --test
+if [[ $? != 4 ]]; then
+    echo 'getopt (enhanced) must be installed to parse CLI flags. You have the non-enhanced version.'
+    exit 1
+fi
+
+# Parse flags
+DO_FIX=0
+WITH_MD=0
+WITH_PHP=0
+WITH_JS=0
+OPTS=$(getopt -o 'fmpj' --long 'fix,with-md,with-php,with-js' -n 'ss-docs-lint' -- "$@")
+eval set -- "$OPTS"
+# getopt adds a "--" after all of the options it has found,
+# so "while true" is safe because we will always hit that break.
+while true; do
+    case "$1" in
+        '-f'|'--fix')
+            DO_FIX=1
+            shift 1
+            continue
+        ;;
+        '-m'|'--with-md')
+            WITH_MD=1
+            shift 1
+            continue
+        ;;
+        '-p'|'--with-php')
+            WITH_PHP=1
+            shift 1
+            continue
+        ;;
+        '-j'|'--with-js')
+            WITH_JS=1
+            shift 1
+            continue
+        ;;
+        # Anything after "--" on its own can be ignored - it will all be args from here on
+        '--')
+            shift
+            break
+        ;;
+        # This should never be reached (getopt doesn't give us args in this loop) - but throw an error just in case
+        *)
+            echo "Internal error! got '$1'" >&2
+            exit 1
+        ;;
+    esac
+done
+
+# lint everything if we're not told what to lint
+if [[ $WITH_MD == 0 && $WITH_PHP == 0 && $WITH_JS == 0 ]]; then
+    WITH_MD=1
+    WITH_PHP=1
+    WITH_JS=1
+fi
+
+# If there's any arguments, the first arg should be the module directory
+MODULE_DIR=$1
+if [[ -z $MODULE_DIR ]]; then
+    MODULE_DIR=$ORIG_DIR
+fi
+
+# Get and validate the config file path
+CONFIG_FILE="${MODULE_DIR}/.ss-docs-lint"
+if ! [[ -f $CONFIG_FILE ]]; then
+    echo "No '.ss-docs-lint' file found at ${MODULE_DIR}"
+    exit 1
+fi
+
+# Get and validate the documentation directory
+DOCS_DIR="${MODULE_DIR}/$(head -n 1 $CONFIG_FILE)"
+if ! [[ $DOCS_DIR == *'/' ]]; then
+  echo "${DOCS_DIR} must end with / in the .ss-docs-lint file"
+  exit 1
+elif ! [[ -d $DOCS_DIR ]]; then
+  echo "${DOCS_DIR} is not a directory"
+  exit 1
+fi
+
+PHP_LINT="${COMPOSER_RUNTIME_BIN_DIR}/mdphpcs"
+
+# change to the docs directory, or else yarn may stop us from linting the files we want to lint
+cd $DOCS_DIR
+
+# Prepare a cleanup function
+cleanup () {
+    rm package.json >/dev/null 2>&1
+    rm yarn.lock >/dev/null 2>&1
+    rm yarn-error.log >/dev/null 2>&1
+    rm .markdownlint-cli2.mjs >/dev/null 2>&1
+    rm .markdownlint.yml >/dev/null 2>&1
+    rm phpcs.xml >/dev/null 2>&1
+    rm .eslintrc.js >/dev/null 2>&1
+    rm -r node_modules/ >/dev/null 2>&1
+}
+
+# If linting markdown or js we need to install the yarn dependencies.
+# These have to be in the directory that holds the docs, or a parent directory of it,
+# otherwise the linting tools will refuse to lint for us.
+if [[ $WITH_MD == 1 || $WITH_JS == 1 ]]; then
+    cp "${DOCS_LINT_DIR}/package.json" package.json
+    yarn install
```

Comment:
I tried to avoid this, I really did, but the js linters just refused to lint anything that wasn't a child directory of the one with the `package.json` file.